code/game/objects/items/holotool.dm

```diff
@@ -0,0 +1,83 @@
+
+
+/obj/item/holotool
+	name = "experimental holotool"
+	desc = "A highly experimental holographic tool projector."
+	icon = 'icons/obj/tools.dmi'
+	icon_state = "holotool"
+	slot_flags = SLOT_BELT
+	usesound = 'sound/items/pshoom.ogg'
+	lefthand_file = 'icons/mob/inhands/equipment/tools_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/equipment/tools_righthand.dmi'
+	actions_types = list(/datum/action/item_action/change_tool, /datum/action/item_action/change_ht_color)
+
+	var/on = FALSE
+	var/obj/item/current_tool = 0
+	var/current_color = "#48D1CC" //mediumturquoise
+
+/obj/item/holotool/proc/AddTool(typee, namee)
+	var/obj/item/WR = new typee(src)
+	WR.forceMove(src)
+	WR.name = namee
+	WR.usesound = usesound //use the same sound as we do
+	WR.toolspeed = 0.5
+
+/obj/item/holotool/proc/AddTools()
+	AddTool(/obj/item/wrench, "holo-wrench")
+	AddTool(/obj/item/screwdriver, "holo-screwdriver")
+	AddTool(/obj/item/wirecutters, "holo-wirecutters")
+	AddTool(/obj/item/weldingtool/largetank, "holo-welder")
+	AddTool(/obj/item/crowbar, "holo-crowbar")
+	AddTool(/obj/item/device/multitool, "holo-multitool")
+
+/obj/item/holotool/Initialize()
+	. = ..()
+	//create and rename tools
+	AddTools()
+
+/obj/item/holotool/attack_self(mob/living/user)
+	to_chat(user, "<span class='notice'>You turn [src] [on ? "off" : "on"].</span>")
+	on = !on
```

Comment:
why do you even have this toggle.

Reply:
for the chat message

Reply:
that doesn't answer why the toggle exists at all.